core/src/main/java/aima/core/search/basic/support/BasicProblem.java

```diff
@@ -1,24 +1,26 @@
 package aima.core.search.basic.support;
 
-import java.util.List;
-
-import aima.core.search.api.ActionsFunction;
-import aima.core.search.api.GoalTestPredicate;
+import aima.core.search.api.BidirectionalProblem;
 import aima.core.search.api.NondeterministicProblem;
 import aima.core.search.api.OnlineSearchProblem;
 import aima.core.search.api.Problem;
+import aima.core.search.api.ActionsFunction;
 import aima.core.search.api.ResultFunction;
-import aima.core.search.api.ResultsFunction;
+import aima.core.search.api.GoalTestPredicate;
 import aima.core.search.api.StepCostFunction;
+import aima.core.search.api.ResultsFunction;
+import java.util.List;
+
 
 /**
  * Basic implementation of the Problem, NondeterministicProblem, and
  * OnlineSearchProblem interfaces.
  *
  * @author Ciaran O'Reilly
  */
-public class BasicProblem<A, S> implements Problem<A, S>, NondeterministicProblem<A, S>, OnlineSearchProblem<A, S> {
+public class BasicProblem<A, S> implements Problem<A, S>, NondeterministicProblem<A, S>, OnlineSearchProblem<A, S>, BidirectionalProblem<A, S> {
 	private S initialState;
```

Comment:
Declare implements BidirecationalProblem after Problem<A, S>, i.e.:

implements Problem<A, S>, Bidirectional....

so that is matches the order these interfaces are introduced in the book.